src/munge.ts

```diff
@@ -69,50 +70,75 @@ export function removeCodec(sdpOrAv: Sdp | AvMediaDescription, codecName: string
 
 /**
  * Retain specific codecs, filtering out unwanted ones from the given SDP or audio/video media
- * description.
+ * description. The provided predicate should take in a single {@link codecInfo}, and only codecs
+ * for which the predicate returns true will be retained.
  *
- * Note: Done this way because of a feature not implemented in all browsers, currently missing in
- * Firefox. Once that is added we can use `RTPSender.getCapabilities` and filter those to call
- * with `RTCRtpTransceiver.setCodecPreferences` instead of doing this manually.
+ * Note: Done this way because of a feature that was only recently implemented in all browsers,
+ * previously missing in Firefox. You can also use `RTPSender.getCapabilities` and filter those to
+ * call with `RTCRtpTransceiver.setCodecPreferences` instead of doing this manually.
  *
  * @param sdpOrAv - The {@link Sdp} or {@link AvMediaDescription} from which to filter codecs.
- * @param allowedCodecNames - The names of the codecs that should remain in the SDP.
+ * @param predicate - A function used to determine which codecs should be retained.
+ * @returns A boolean that indicates if some codecs have been filtered out.
  */
 export function retainCodecs(
   sdpOrAv: Sdp | AvMediaDescription,
```

Comment:
I wonder, do we ever use this with an entire SDP?  Retaining codecs across media types doesn't make much sense, not sure why I supported that (or if it was even me, I guess)

I guess the predicate version it makes a bit more sense since then it can detect audio vs video.

Reply:
That's a good question. I took a look at the history for this file. It seems like `removeCodec` was the original helper function which took in an SDP (and it's still in this file, though I don't see it being used anywhere). Then we added the current `retainCodecs` and `retainCandidates` functions... and I'm not sure why I had it accept both? Probably because it made sense to take in a media description for those, but `removeCodec` took in an SDP, so I compromised and just had all of them take both.

Reply:
Ah ok, yeah I think `removeCodec` made sense with an SDP because it would remove all instances of those codecs from all mlines, so it was easy to throw both video and audio codecs in there.  With 'retain' though, for codecs, I'm not sure it makes sense (transport types still accepting an entire sdp does, though)

Reply:
@bbaldino Will it be logical to keep the API consistent between these functions and more flexible? Or is your vision always to retain codecs only from `AvMediaDescription` when working with SDP?